Add load-dependent data-based HP and chiller models

AntoineGautier opened the pull request:
This addresses https://github.com/lbl-srg/modelica-buildings/issues/4169

JayHuLBL commented on Buildings/Fluid/Chillers/ModularReversible/BaseClasses/RefrigerantCycleHeatRecovery.mo:
```diff
@@ -0,0 +1,159 @@
+  "Refrigerant cycle model for cooling-only or heat recovery chiller models"
+  extends Buildings.Fluid.HeatPumps.ModularReversible.BaseClasses.PartialModularRefrigerantCycle(
+    use_rev=false);
+  parameter Boolean have_switchover=false
```
may change to `have_switchOver`?

AntoineGautier replied:
The naming was chosen for consistency with the same parameter in `Buildings.Fluid.Chillers.BaseClasses.PartialElectric`.

JayHuLBL commented on Buildings/Fluid/Chillers/ModularReversible/BaseClasses/RefrigerantCycleHeatRecovery.mo:
```diff
@@ -0,0 +1,159 @@
+      coordinateSystem(
+        preserveAspectRatio=false)),
+    Documentation(
+      revisions="<html>
```
This is a new model. We would need to have the new revision note.

AntoineGautier replied:
Addressed

JayHuLBL commented on Buildings/Fluid/Chillers/ModularReversible/Data/TableData2DLoadDep/Generic.mo:
```diff
@@ -0,0 +1,47 @@
+</html>",
+      revisions="<html><ul>
+  <li>
+    <i>October 2, 2022</i> by Fabian Wuellhorst:<br/>
```
Start the new release note.

JayHuLBL commented on Buildings/Fluid/HeatPumps/ModularReversible/RefrigerantCycle/BaseClasses/Validation/TableData2DLoadDep.mo:
```diff
@@ -0,0 +1,303 @@
+  connect(coo.y, chiHeaSupLvg.coo) annotation (Line(points={{-58,20},{-22,20},{-22,
+          -13},{-2,-13}},
+                       color={255,0,255}));
+
```
remove empty lines

AntoineGautier replied:
Addressed

JayHuLBL reviewed (commented):
@AntoineGautier Please see the inline comments, which are minor.

AntoineGautier commented on Buildings/Fluid/Chillers/ModularReversible/TableData2DLoadDep.mo:
```diff
@@ -0,0 +1,265 @@
+    displayUnit="degC")
+    "Temperature setpoint"
+    annotation (Placement(transformation(extent={{-180,20},{-140,60}}),
+      iconTransformation(extent={{-142,0},{-102,40}})));
```
I've moved the connector's icon. (Note that `ySet` is actually not instantiated in this model due to `final use_busConOnl=true`.)

AntoineGautier commented:
@mwetter Could you have a final look on this PR? (Note the joint issue #4261 that I've opened to include performance data sets based on the records for the polynomial chiller model.)

mwetter reviewed (commented):
This is a great addition. I made minor cosmetic changes. Could you please also
- [x] Update the library release notes.
- [ ] Mention the new models in the heat pump user guide.
- [ ] Merge changes that would be needed to IBPSA if they affect the existing models.

AntoineGautier commented:
@mwetter I've made some minor changes in e2cbb7e855. 

- I've moved the declaration of `P_min "Minimum power when system is enabled with compressor cycled off"` from the data record to the main model `Buildings.Fluid.HeatPumps.ModularReversible.TableData2DLoadDep`: this is because this parameter is used as is, and not scaled as other variables computed from the performance data provided via the record. In this respect, it better goes along with the nominal capacity specified in the main model.
- I've added the declaration of `tabNamQ "Table names with heat flow rate data"` and `tabNamP "Table names with power data"` to the data record: this is because these names must match the ones used in the table specified in the data record, so it seems like a good place to specify them. Previously, one could only modify these names by modifying the bindings `refCyc.refCycHeaPum*.calQUseP.tabNam*` deeply nested in the model composition.

AntoineGautier commented:
@mwetter This is ready on my end. You can merge it.
